build_tools/cmake/iree_check_test.cmake

```diff
@@ -23,7 +23,7 @@ function(iree_bytecode_module_for_iree_check_test_and_friends)
     ${ARGN}
   )
 
-  if(ANDROID)
+  if(ANDROID AND NOT _RULE_FLAGS MATCHES "iree-llvm-target-triple")
```

Comment:
This just skips appending a target triple if one is already provided, so when the wasm triple is set this doesn't try to infer an additional Android triple.

I think a better (or maybe complementary?) fix here is probably to skip building/defining `iree_bytecode_module` targets for the CUDA/Wasm/WebGPU/Metal compiler target backends entirely when cross compiling for Android.

Reply:
Oof this just points out how messed up our testing is...

Reply:
I took a peek at `process_target_cpu_features()` while in this file and it spooked me quite a bit O_O. This is tame in comparison.